### ISSUE TITLE
is this thing still being developed or is it abandoned?...

### ISSUE THREAD

**Issue by razvanc-r** [koajs/koala#19]

**Comment by jonathanong** [comment]
sorry, kind of slow on open source now. you could still use it - it's just a combination of middleware and extensions. you could use the middleware and extensions directly (which i actually do and recommend more because it's more stable that way)

still waiting on koa@2 and http2 is too early to be useful since nginx doesn't support HTTP2 push. thus, it's kind of in limbo

if anyone would like to maintain this, that would be great!

**Comment by razvanc87** [comment]
I know that koala is just conveniently gathering up middleware and
extensions :). I don't know exactly what I was hoping for really. I guess
for beginners (like myself) it's nice to have something to start from
instead of doing all the research about modules and extensions since there
are so many of them out there. Thanks for the replay.

**Comment by pocesar** [comment]
dream coming true @jonathanong? https://www.nginx.com/blog/nginx-1-9-5/ :)

**Comment by jonathanong** [comment]
> HTTP/2’s ‘Server Push’ feature is not supported in this release.

:( might take a while

**Comment by pocesar** [comment]
oops, my dynamic reading has failed me once more ;P

**Comment by nicolasmccurdy** [comment]
Would it make sense to implement support for Koa 2 before HTTP2 support is implemented? I feel like this would be really useful for Koa 2 in the meatime.

**Comment by jonathanong** [comment]
@nicolasmccurdy correct

**Comment by jonathanong** [comment]
also, i've found that cloud flare uses link tags vs. pushing from the origin, so that lowers my priority on HTTP push a little

**Comment by jonathanong** [comment]
btw if anyone wants to maintain this for the time being, let me know. i'm not sure when i'll be back on this.

**Comment by nicolasmccurdy** [comment]
I'm new to Koa but I'm excited about its direction and I've used Koala a little bit in the past. Do you know roughly how complex it would be to migrate to Koa 2? It seems like it's mostly updating middleware and dealing with dependency changes. I'd be happy to help depending on how much time it takes.

**Comment by jonathanong** [comment]
just updating middleware. the middleware itself might need updating though.

**Comment by nicolasmccurdy** [comment]
Let's move Koa 2 discussion to #17.

**Comment by nfantone** [comment]
@nicolasmccurdy You _may_ me interested in checking out [mu-koan](https://github.com/nfantone/mu-koan), a personal project. It's WIP, for now. It's like Koala, but for Koa 2, aimed at building JSON REST APIs. It adds some other bits like automatic routing, logging and graceful shutdown.

**Comment by nicolasmccurdy** [comment]
@nfantone That seems cool! Have you considered upstreaming some of your work? It seems like it could be useful to this project if the routing support was removed.

**Comment by nfantone** [comment]
@nicolasmccurdy Sure! We could ping-pong some ideas and try to bring together the best of the two projects.

**Comment by jonathanong** [comment]
@nfantone let me know if you're interesting in contributing here!

**Comment by doug-wade** [comment]
@jonathanong lemme know if you still need another maintainer; I can pick this up in ~2 weeks.

**Comment by nfantone** [comment]
@doug-wade I could also give you a hand here. I think this project has potential but it has seen brighter days. (cc. @jonathanong)

**Comment by jonathanong** [comment]
@doug-wade @nfantone added both of u

**Comment by nfantone** [comment]
@jonathanong Thanks, man! 

@doug-wade What are your thoughts on the direction we should give to `koala`? TBH, I am not a big fan of re-implementing things that are already modularized and available out there. For instance, all the middlewares [here](https://github.com/koajs/koala/tree/master/lib/middleware) can be replaced with external npm modules.

Also, I'd like this to be more of a _wrapper_/_enhancer_ around a Koa app, rather than *being* a Koa app itself. That way, we could declare `"koa"` as a `"peerDependency"` and inverse control over the Koa instance itself and not restrict users to an anchored version.

I have followed and implemented a similar approach over at [a personal project](https://github.com/nfantone/mu-koan).

**Comment by doug-wade** [comment]
@jonathanong I've filed #30, which seems like a great idea.  I'd be interested in hearing more about the second one, so I set up a [gitter channel]() for us to discuss further what you're thinking.

We've added 2 new maintainers @razvanc-r, which I think is enough to assert the project isn't abandoned.  It might take us a bit to get into a regular release cadence, though.

**Comment by nickmccurdy** [comment]
Please check out the new [Contributing](https://github.com/koajs/koala/blob/master/CONTRIBUTING.md) docs if you'd like to pitch in. 😄
